samza-core/src/main/java/org/apache/samza/coordinator/staticresource/StaticResourceJobCoordinator.java

```diff
@@ -234,13 +262,33 @@ private void prepareWorkerExecution(JobModel jobModel, JobCoordinatorMetadata ne
     }
   }
 
+  /**
+   * Wrapper around {@link MetadataResourceUtil} constructor so it can be stubbed during testing.
+   */
   @VisibleForTesting
   MetadataResourceUtil metadataResourceUtil(JobModel jobModel) {
     return new MetadataResourceUtil(jobModel, this.metrics, this.config);
   }
 
+  /**
+   * Wrapper around {@link DiagnosticsUtil#buildDiagnosticsManager} so it can be stubbed during testing.
+   */
+  @VisibleForTesting
+  Optional<DiagnosticsManager> buildDiagnosticsManager(String jobName,
+      String jobId, JobModel jobModel, String containerId, Optional<String> execEnvContainerId, Config config) {
+    return DiagnosticsUtil.buildDiagnosticsManager(jobName, jobId, jobModel, containerId, execEnvContainerId, config);
+  }
+
   private Set<JobMetadataChange> checkForMetadataChanges(JobCoordinatorMetadata newMetadata) {
     JobCoordinatorMetadata previousMetadata = this.jobCoordinatorMetadataManager.readJobCoordinatorMetadata();
     return this.jobCoordinatorMetadataManager.checkForMetadataChanges(newMetadata, previousMetadata);
   }
+
+  private static void quietlyStop(DiagnosticsManager diagnosticsManager) {
+    try {
+      diagnosticsManager.stop();
```

Comment:
Since you are making some changes related to diagnostics manager, I noticed the following within the `stop()` method

```
scheduler.shutdown();

    // Allow any scheduled publishes to finish, and block for termination
    scheduler.awaitTermination(terminationDuration.toMillis(), TimeUnit.MILLISECONDS);

    if (!scheduler.isTerminated()) {
      LOG.warn("Unable to terminate scheduler");
      scheduler.shutdownNow();
    }
    this.systemProducer.stop();
    
    ``
    
    It seems like the `systemProducer` can linger in the event Interruption occurs due to `awaitTermination`. Would you mind fixing this as well to ensure `systemProducer.stop()` is invoked as part of finally?

Reply:
Also noticed that the `shutdownNow()` is never going to get invoked if there is interruption since awaitTermination will throw interrupted exception short circuiting the rest of the method.